src/ast/ast.h

```diff
@@ -993,7 +989,9 @@ class ast_translation;
 
 class ast_table : public chashtable<ast*, obj_ptr_hash<ast>, ast_eq_proc> {
 public:
-    ast_table() : chashtable({}, {}, 512 * 1024, 8 * 1024) {}
+    // Cache-optimized parameters: start smaller for better cache locality
+    // and allow growth as needed. This improves performance for typical workloads.
+    ast_table() : chashtable({}, {}, 16 * 1024, 2 * 1024) {}
```

Comment:
This is a bad change. We need to start with a large table so we don't pay the price of frequent (and expensive) re-hashes.